### PR TITLE
Add mesos reqs.  Addresses #2054 and #2070.

### PR DESCRIPTION
**Pull request by DailyDreaming**
Add mesos reqs to the three cloud reqs.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -77,12 +77,15 @@ def runSetup():
             'docker==2.5.1'],
         extras_require={
             'mesos': mesos_reqs,
```
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply setup.py]: Mesos makes sense on its own, since it is an independent batch system.  I can change the docs and the cloud configs.

**Comment by DailyDreaming** [comment]
Everything should be up-to-date.  I checked the docs more comprehensively and really only the install instructions needed any changes.

**Comment by DailyDreaming** [comment]
Flaky test:

- src.toil.test.src.fileStoreTest.CachingFileStoreTestWithGoogleJobStore.testFileStoreExportFile
- http://jenkins.cgcloud.info/job/toil-pull-requests/3660/

**Comment by DailyDreaming** [comment]
Jenkins, test this please.

File: docs/gettingStarted/install.rst [hunk 2]
```diff
@@ -116,8 +114,7 @@ Here's what each extra provides:
 |                | * :ref:`libffi headers and library <libffi-dev>`           |
 +----------------+------------------------------------------------------------+
 | ``cwl``        | Provides support for running workflows written using the   |
-|                | `Common Workflow Language`_. This extra has no native      |
-|                | dependencies.                                              |
+|                | `Common Workflow Language`_.                               |
```
Comment (ejacox) [inline docs/gettingStarted/install.rst]: Could you add HTCondor to this list, but note that it is not in the 'all' extra.

File: setup.py [hunk 3]
```diff
@@ -77,12 +77,15 @@ def runSetup():
...
+            'azure': azure_reqs + 
+                     mesos_reqs,
+            'google': google_reqs + 
+                      mesos_reqs,
             'encryption': encryption_reqs,
-            'google': google_reqs,
             'cwl': cwl_reqs,
-            'htcondor': htcondor_reqs,
             'all': mesos_reqs +
                    aws_reqs +
```
Comment (ejacox) [inline setup.py]: Remove htcondor from all.

**Comment by DailyDreaming** [comment]
@ejacox Should be fixed.

**Comment by DailyDreaming** [comment]
Probable flaky tests:
- src.toil.test.provisioners.aws.awsProvisionerTest.AWSAutoscaleTest.testSpotAutoScale
- src.toil.test.provisioners.aws.awsProvisionerTest.AWSAutoscaleTestMultipleNodeTypes.testAutoScale
- http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/1998/

**Pull request merged**